Functions for static matching

davidsantiagoquevedo opened the pull request:
This PR introduces two functions for matching:

* `match_cohort()` matches the cohort based on nearest and exact characteristics. It uses matching algorithms implemented in MatchIt 

* `censor_match()` imputes the censoring date to matched couples

### Notes: 
* Both functions include testing scripts. To keep 100% coverage of the package
* In commit https://github.com/epiverse-trace/vaccineff/commit/47d70e2c2d0ef274201c72c5595877c2cd41b2bf a refactorization of `censor_match()` was included to correct a spelling mistake.

GeraldineGomez commented on R/coh_matching.R:
```diff
@@ -0,0 +1,204 @@
+    distance = "glm"
+  )
+  match <- MatchIt::match.data(matchit, distance = "prop.score")
+  names(match) <- gsub(x = names(match),
```
Candidate code to develop a function in dev-utils

davidsantiagoquevedo replied:
I just opened a [new branch](https://github.com/epiverse-trace/vaccineff/tree/dev-utils) to work on dev-utils

davidsantiagoquevedo commented:
Final edits:

* Added control to guarantee that the censoring date of matched couples is always greater than the outcome date.
* Added input checks to get_censoring_date_match().
* Renamed censor_cohort() as get_censoring_date_match() to make it easier for users to understand.
* Refactored match_cohort to unify input for caliper and nearest.
